src/app/features/charts/components/score-barchart/score-barchart.component.ts

```diff
@@ -45,26 +45,32 @@ export class ScoreBarChartComponent implements AfterViewInit, OnChanges, OnDestr
   constructor(private helperService: HelperService) {
   }
 
-  ngOnChanges(): void {
-    if (this.initialized) {
-      // if chart has already been initialized, this means the data has changed
-      this.createChart();
+  ngOnChanges(changes: SimpleChanges): void {
+    if (
+      (changes.data && !changes.data.firstChange) ||
+      (changes.score && !changes.score.firstChange) ||
+      (changes.barColor && !changes.barColor.firstChange)
+    ) {
+      if (this.score === null) {
+        this.clearChart();
+        this.hideChart();
+      } else {
+        this.clearChart();
+        this.showChart();
+        this.createChart();
+      }
     }
   }
 
-  ngOnDestroy(): void {
-    this.destroyChart();
-  }
-
   ngAfterViewInit(): void {
-    if (!this.chartRef.nativeElement) {
-      return;
-    }
-
-    if (!this.data)
-      return;
+    if (this.score === null)
```

Comment:
if the score is null, we need to hide it on initial load, otherwise we create the chart.